app/models/collection_participant.rb

```diff
@@ -57,4 +59,14 @@ def user_allowed_to_promote?(user, role)
     (role == MEMBER || role == NONE) ? self.collection.user_is_maintainer?(user) : self.collection.user_is_owner?(user)
   end
 
+  def reindex_collection
+    is_assigned_as_maintainer = MAINTAINER_ROLES.include?(participant_role)
+    participant_change = previous_changes[:participant_role]
+    is_removed_as_maintainer = participant_change.present? && MAINTAINER_ROLES.include?(participant_change.first) && MAINTAINER_ROLES.exclude?(participant_change.second)
+    return unless is_assigned_as_maintainer || is_removed_as_maintainer
+
+    ids = [collection_id]
+    ids.push(collection.children.pluck(:id)) if collection.children.any?
```

Comment:
Oops, yeah, that makes sense. Should I flatten it out?

Reply:
Maybe just set `ids = [collection_id] + collection.children.pluck(:id)`? That's similar to the way child IDs are handled in `Collection#all_items`:
https://github.com/otwcode/otwarchive/blob/b5bf2e7a5cd362d1e350f2be087e8049721a2484/app/models/collection.rb#L281-L283
Alternatively, maybe `+=` or `concat` instead of `push`? `flatten` would work as well, but it does involve an extra step.